components/activecampaign/sources/common-webhook.js

```diff
@@ -0,0 +1,49 @@
+const activecampaign = require("../activecampaign.app.js");
+const common = require("./common.js");
+
+module.exports = {
+  ...common,
+  props: {
+    ...common.props,
+    http: "$.interface.http",
+    sources: { propDefinition: [activecampaign, "sources"] },
+  },
+  methods: {
+    isRelevant(body) {
+      return true;
+    },
+  },
+  hooks: {
+    async activate() {
+      const sources =
+        this.sources.length > 0
+          ? this.sources
+          : this.activecampaign.getAllSources();
+      const hookData = await this.activecampaign.createHook(
+        this.getEvents(),
+        this.http.endpoint,
+        sources
+      );
+      this.db.set("hookId", hookData.webhook.id);
+    },
+    async deactivate() {
+      await this.activecampaign.deleteHook(this.db.get("hookId"));
+    },
+  },
+  async run(event) {
+    const { body } = event;
+    if (!body) {
+      return;
+    }
+
+    if (!this.isRelevant(body)) return;
+
+    const dateTime = new Date(body.date_time);
+    const { id, summary } = await this.getMeta(body);
+    this.$emit(body, {
+      id,
+      summary,
+      ts: dateTime.getTime(),
+    });
```

Comment:
Minor: why not have the `getMeta` method return the `ts` as well, since it's part of the metadata?

In the event source...
```js
module.exports = {
  methods: {
    getMeta(body) {
      // Extract ID and summary...
      const { date_time: dateTimeIso } = body;
      const ts = Date.parse(dateTimeIso);
      return {
        id,
        summary,
        ts,
      };
    };
  },
};
```

Then in `common-webhook`:
```js
module.exports = {
  async run(event) {
    // Initial checks...
    const meta = await this.getMeta(body);
    this.$emit(body, meta);
  },
};
```

I see the potentially duplicated code in the different `getMeta` implementations, but that's only accidental, and removing that duplicate logic introduces other downsides, like having different responsibilities of `getMeta` depending on the event source we're looking at (an example is `new-automation-webhook`, which in fact returns the complete metadata information of the event).

Again, this is a **minor** comment and I'm not blocking the PR because of this, just food for though 🙂